### DISCUSSION TITLE
Way to define ipv6 (ip6) with scripts?

### DISCUSSION THREAD

**Discussion by popallo** [tteck/Proxmox#2545] [Q&A]
Hi everyone, new here :). 

I was wondering if, by any chance, it is possible to define the IPv6 of a container during the script build process?

I don't think it's possible at the moment, but maybe I'm missing something?

And I didn't find a discussion on this subject or I also missed it (and I apologize if this is the case, don't hesitate to delete this discussion).

Thank you in advance for your insight ;).

**Comment by tteck** [comment]
Currently, the scripts are not configured to assign IPv6 addresses.

**Reply by popallo** [reply]
Thanks for the confirmation, so I'm not crazy or illiterate :). I was afraid I had missed an optional parameter or other magical manipulation to obtain the IPv6 menu.

**Reply by tteck** [reply]
I began working on it some time ago, but it's been a while since then.

```
# Function to generate random blocks
generate_blocks() {
    local blocks=""
    for ((i=0; i<$1; i++)); do
        blocks+=":$(openssl rand -hex 2)"
    done
    echo "$blocks"
}
# Generate the IPv6 address and remove double colons if they occur in succession
ipv6_address="2023$(generate_blocks 2)::$(generate_blocks 3)" 
ipv6_address=$(echo "$ipv6_address" | sed 's/::/:/')

echo "IPv6 address: $ipv6_address"
```

This will create a random IPv6 address with a static prefix '2023' (I just used the year), two randomly generated blocks, a double colon (representing consecutive blocks of zeros), and three more randomly generated blocks to form a valid IPv6 address. e.g. `2023:0228:31f4::5d48:c56d:5da6`

I need to incorporate the snippet and make modifications to the `build.func` script. Perhaps someday.

**Comment by popallo** [comment]
Ok thanks for the feedback.

I was mainly thinking of the possibility of determining manually, in the same way as IPv4, IPv6 and the gateway.

This may only concern me but I assign an external IPv6 to my containers, to the extent that we have a block of IPv6 /64 provided on the remote servers provided by the different service providers.

**Reply by tteck** [reply]
I understand. With IPv6, I thought that displaying a random address would illustrate the format, allowing you to manually configure it. However, I can envision that users unfamiliar with IPv6 might encounter difficulties, which is why progress stalled.

**Reply by popallo** [reply]
I imagined a "choice" system a bit like the system choice (debian | ubuntu), leading to a screen similar to that of the IPv4 configuration.
That way, those who don't know IPv6 don't venture into it :D
We're all going to have to switch to IPv6 at some point, including github.com ;)
